### PR TITLE
chore: Merge v5 changes to main

### PR DESCRIPTION
**Pull request by stocaaro**
#### Description of changes
chore: Merge v5 changes to main

PR Generated using the following list of commands:
```sh
git co main
git merge next-major-version/5
git co next-major-version/5 ./
git rm -rf ./packages/amplify-ui
git rm -rf ./packages/amplify-ui-angular
git rm -rf ./packages/amplify-ui-components
git rm -rf ./packages/amplify-ui-react
git rm -rf ./packages/amplify-ui-storybook
git rm -rf ./packages/amplify-ui-vue
git rm -rf ./packages/aws-amplify-angular
git rm -rf ./packages/aws-amplify-react
git rm -rf ./packages/aws-amplify-vue
git add .
git commit --allow-empty --no-verify -am 'chore: Merge v5 changes to main'
```

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jimblanc — APPROVED** [review]
🥳

**Review by elorzafe — APPROVED** [review]
Looks to me! 🎉 

Thanks @stocaaro

**Review by AllanZhengYP — APPROVED** [review]

**Review by cwomack — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -369,8 +369,7 @@ jobs:
           # https://github.com/TryGhost/node-sqlite3#source-install
           command: |
             cd packages/datastore-storage-adapter
-            npm install --build-from-source
-            rm -rf node_modules/@aws-amplify node_modules/@aws-sdk
```
Comment (david-mcafee) [inline .circleci/config.yml:373]: Why is this being removed?

**Review by svidgen — APPROVED** [review]

**Review by wlee221 — APPROVED** [review]
This LGTM, based on:
(1) UI team's canary builds against `next` has been successful
(2) legacy ui components are good to be removed

cc @calebpollman in case I missed something

**Review by david-mcafee — COMMENTED** [review]
LGTM!

**Pull request merged**